storm-core/src/jvm/backtype/storm/security/auth/authorizer/SimpleACLAuthorizer.java

```diff
@@ -50,6 +50,7 @@
     protected Set<String> _supervisors;
     protected IPrincipalToLocal _ptol;
     protected IGroupMappingServiceProvider _groupMappingProvider;
+    protected ImpersonationAuthorizer _impersonationAuthorizer;
```

Comment:
I don't really like the idea of having all authorizers have to be updated to support the impersonation code.  I would much rather have the code spliced in at a much higher level that cannot be turned off.  Otherwise a custom IAuthorizer instance that does not know to update after this change is now wide open for anyone to impersonate anyone else.  The logging changes should also probably happen at the same level.